### DISCUSSION TITLE
[Bug]: How to set max epochs parameter ?

### DISCUSSION THREAD

**Discussion by shanmugamani1023** [open-edge-platform/anomalib#1962] [Q&A] [answered]
### Describe the bug

from anomalib.data import Folder

 

i tried to train model using custom dataset using following code and 
# Create the datamodule
datamodule = Folder(
    name="bottle",
    root="datasets/bottle",
    normal_dir="good",
    abnormal_dir="bad",
    task="classification",
    image_size=(512,512)
  )

# Setup the datamodule

datamodule.setup()

and 

i used efficientad model for transfer learning ,
from anomalib.models import EfficientAd
from anomalib.engine import Engine

# Create the model and engine
model = EfficientAd()
engine = Engine(task="classification")
engine.train(datamodule=datamodule, model=model)

but , i dont know how to set my custom epoch value , if i give that max_epochs parameter in anomalib\configs\model\efficient_ad.yaml ,it didnt work, how to change that?
![anamolin](https://github.com/openvinotoolkit/anomalib/assets/119931738/a619f156-8911-4e90-9af4-67a3cb8ddf5d)

### Dataset

Other (please specify in the text field below)

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

from anomalib.data import Folder

 

i tried to train model using custom dataset using following code and 
# Create the datamodule
datamodule = Folder(
    name="bottle",
    root="datasets/bottle",
    normal_dir="good",
    abnormal_dir="bad",
    task="classification",
    image_size=(512,512)
  )

# Setup the datamodule

datamodule.setup()

and 

i used efficientad model for transfer learning ,
from anomalib.models import EfficientAd
from anomalib.engine import Engine

# Create the model and engine
model = EfficientAd()
engine = Engine(task="classification")
engine.train(datamodule=datamodule, model=model)

but , i dont know how to set my custom epoch value , if i give that max_epochs parameter in anomalib\configs\model\efficient_ad.yaml ,it didnt work, how to change that?
![anamolin](https://github.com/openvinotoolkit/anomalib/assets/119931738/a619f156-8911-4e90-9af4-67a3cb8ddf5d)

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04]
- Python version: [e.g. 3.10.0]
- Anomalib version: [e.g. 0.3.6]
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

null

### Screenshots

![anamolin](https://github.com/openvinotoolkit/anomalib/assets/119931738/e413509f-202f-456e-a298-6baaf3ef1d4b)

### Pip/GitHub

pip

### Configuration YAML

```yaml
null
```

### Logs

```shell
null
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by samet-akcay** [comment]
You could set it when you instantiate `Engine`

```python
model = EfficientAd()
engine = Engine(task="classification", max_epochs=1000)
engine.train(datamodule=datamodule, model=model)

but , i dont know how to set my custom epoch value , if i give that max_epochs parameter in anomalib\configs\model\efficient_ad.yaml ,it didnt work, how to change that?
```

**Reply by shanmugamani1023** [reply]
i tried that in Patchcore ,
# Import the model and engine
from anomalib.models import Patchcore
from anomalib.engine import Engine

# Create the model and engine
model = Patchcore()
engine = Engine(task="classification",max_epochs=2)

# Train a Patchcore model on the given datamodule
engine.train(datamodule=datamodule, model=model)
but it epoch runs only one time?
![Uploading padim.PNG…]()

**Reply by blaz-r** [reply, marked as answer]
Patchcore isn't trained using backpropagation, so it runs only a single epoch in all cases. This is expected. If you want to learn more, check the [Patchcore readme](https://github.com/openvinotoolkit/anomalib/tree/main/src/anomalib/models/image/patchcore).

**Reply by shanmugamani1023** [reply]
thank you.
